Projects/UOContent/Gumps/ObjectViewer/BaseViewGump.cs

```diff
@@ -0,0 +1,170 @@
+            Display(owner, page);
+        }
+        public virtual void OnLoad()
+        {
+            throw new Exception("OnLoad not overloaded");
+        }
+
+        private void AddCustomizeEntry(int x, int y)
+        {
+            AddImageTiled(x, y, 161, 23, 0xA40);
+            AddImageTiled(x, y, 159, 21, 0xBBC);
+            AddTextEntry(x, y, 156, 21, 0, 0, "");
+        }
+        private void AddCustomizeLabel(int x, int y, string Name)
+        {
+            AddImageTiled(x, y, 161, 23, 0xA40);
+            AddImageTiled(x, y, 159, 21, 0xBBC);
+            AddLabel(x, y, 0, Name);
+        }
+
+        private void Display(Mobile owner, int page = 0)
+        {
+            _from = owner;
+            Page = page;
+
+            var bodyHeight = (HeightPerItem * perPage) + TopHeight;
+            _totalPage = (Collection.Count / perPage);
+
+            if (Collection.Count % perPage == 0)
+                _totalPage--;
+
+            AddBackground(0, 0, 290, bodyHeight + BottomHeight, BackGumpID);
+            AddHtml(0, 20, 290, 20, $"<BASEFONT COLOR=#F4F4F4>  <CENTER>{Name}</CENTER> Editor </BASEFONT>");
+            AddCustomizeEntry(55, 60);
+            AddButton(90, 90, 5204, 5205, GetButtonID(1, 1));
+
+            if (page > 0) AddButton(110, bodyHeight + 50, 0x15E3, 0x15E7, GetButtonID(2, 1));
+            else AddImage(110, bodyHeight + 50, 0x25EA);
+
+            if (page < _totalPage) AddButton(130, bodyHeight + 50, 0x15E1, 0x15E5, GetButtonID(3, 1));
+            else AddImage(130, bodyHeight + 50, 0x25E6);
+
+            var i = 0;
+            var index = page * perPage;
+            foreach (var item in Collection.Skip(index).Take(perPage))
+            {
+                AddButton(3, (i * HeightPerItem) + TopHeight, 0xFA5, 0xFA7, GetButtonID(4, i + index));
+
+                if (TwoButton)
+                {
+                    AddButton(40, (i * HeightPerItem) + TopHeight, 4023, 4025, GetButtonID(5, i + index));
+                }
+
+                AddCustomizeLabel(80, (i * HeightPerItem) + TopHeight, item.Key);
+                AddButton(250, (i * HeightPerItem) + TopHeight, 0xFA2, 0xFA4, GetButtonID(6, i + index)); // Delete
+                i++;
+            }
+        }
+
+        public virtual bool TwoButton => false;
+        public void Close()
+        {
+            _from.CloseGump<BaseViewGump<T>>();
+        }
+        public virtual void ReOpen()
+        {
+            Close();
+            _from.SendGump(Activator.CreateInstance(this.GetType(), _from, Page) as Gump);
+        }
+
+       
+
+        public override void OnResponse(NetState state, RelayInfo info)
+        {
+            var buttonID = info.ButtonID - 1;
+            var type = buttonID % _typeCount;
+            var index = buttonID / _typeCount;
+
+            switch (type)
+            {
+                case 0: // close
+                case -1: // close
```

Comment:
Consolidate `case 0` and `case -1` into `default` so that all invalid buttons will close the gump?

Reply:
var buttonID = info.ButtonID - 1;  // -1 == 0 
 var type = buttonID % _typeCount;